dev/build_docker_images.sh

```diff
@@ -4,35 +4,49 @@
 #   os
 #   docker_username
 #   docker_password
+#   GITHUB_REF
 
 python_version="3.8"
-tags=""
 version="$(cat version)"
 
+set -ue
+
+git_tag="${GITHUB_REF#refs/tags/v*}"
+if [[ "${git_tag}" != "${version}" ]]; then
+  >&2 echo "ERROR: Mismatch between Git tag ${git_tag} and version file ${version}" 
+  exit 1
+fi
+
 effective_version="${os}_version"
 suffix="${!effective_version}"
-latest="egnyte/gitlabform:latest-${suffix}"
+image_name="egnyte/gitlabform:${version}-${suffix}"
+os_latest="egnyte/gitlabform:latest-${suffix}"
 
-docker pull "${latest}" || echo "no cache is available"
+docker pull "${os_latest}" || echo "no cache is available"
 docker build \
   --build-arg PY_VERSION="${python_version}" \
   --build-arg OS_VERSION="${!effective_version}" \
   --file "dev/${os}.Dockerfile" \
-  --tag "${latest}" .
-tags="$tags ${latest}"
+  --tag "${image_name}" .
+
+tags=( "${image_name}" )
 
-docker tag "${latest}" "egnyte/gitlabform:${version}-${suffix}"
-tags="$tags egnyte/gitlabform:${version}-${suffix}"
+# TODO: something more clever here maybe?
+# TODO: do we need to validate the version?
+if ! [[ "${version}" =~ .+rc.+ ]]; then
+  docker tag "${image_name}" "${os_latest}" 
+  tags+=( "${os_latest}" )
 
-# we treat alpine image as the main one, so it gets the plain "latest" tag
-if [ "${os}" = "alpine" ]
-then
-  docker tag "${latest}" "egnyte/gitlabform:latest"
-  tags="$tags egnyte/gitlabform:latest"
+  # we treat alpine image as the main one, so it gets the plain "latest" tag
+  if [[ "${os}" = "alpine" ]]; then
+    latest=egnyte/gitlabform:latest
+    docker tag "${image_name}" "${latest}"
+    tags+=( "${latest}" )
+  fi
 fi
 
-for image in ${tags}
-do
+echo "Images to be published: ${tags[*]}"
+for image in "${tags[@]}"; do
```

Comment:
[Nit:](https://google.github.io/eng-practices/review/reviewer/standard.html) in these lines you interchange names "tag" and "image" while technically they are different thing. I think that this script would benefit from pulling out only the image name into a separate variable and operate everywhere else on tag and tags. Then only in `docker tag` and `docker push` we would use both.

Reply:
agreed, fixed - can you have another look and see if that now makes more sense?

Reply:
I think that it's much better now, thanks! :)